src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8050,15 +8073,15 @@ bool tryLookupLengthOrCount(string propertyName, out PropertySymbol valid)
                     LookupOptions.Default,
                     originalBinder: this,
                     diagnose: false,
-                    useSiteInfo: ref discardedUseSiteInfo);
+                    useSiteInfo: ref useSiteInfo);
```

Comment:
>useSiteInfo [](http://example.com/codeflow?start=37&length=11)

Consider adding an assert about whether diagnostics and dependencies are being accumulated, or we'll eventually have an issue when a refactor is performed.

Reply:
We've been passing Discard here for ranges too, I followed suit. I can add a prototype comment to revise.